can/common.cc

```diff
@@ -4,10 +4,12 @@ unsigned int honda_checksum(unsigned int address, uint64_t d, int l) {
   d >>= ((8-l)*8); // remove padding
   d >>= 4; // remove checksum
 
+  bool extended = address > 0x7FF;
   int s = 0;
   while (address) { s += (address & 0xF); address >>= 4; }
   while (d) { s += (d & 0xF); d >>= 4; }
   s = 8-s;
+  if (extended){ s += 3; }
```

Comment:
```suggestion
  int s = 0;
  while (address) { s += (address & 0xF); address >>= 4; }
  while (d) { s += (d & 0xF); d >>= 4; }
  s = 8-s;
  if (address > 0x7FF) s += 3; // extended can
```

Reply:
Nice! Thank you

Reply:
The way `address` is used, this won't work as it's always 0 by the time the `3` for the extended id is added. The misra tests in panda actually caught this as I have it the same way there. I've put this back to how it was before.